packages/bart/src/index.ts

```diff
@@ -0,0 +1,626 @@
+      pump_count = 0;
+      balloon_popped = false;
+      cashed_out = false;
+      timed_out = false;
+      explosion_point = Math.floor(Math.random() * (max_pumps - min_pumps)) + min_pumps;
+    }
+        
+    };
+    return singleTrial;
+}
+
+export function createTimeline(jsPsych:JsPsych, { 
+    max_pumps = 20,
+    min_pumps = 1,
+    currency_unit_per_pump = 1, //eg 1 cent per pump
+    num_blocks = 3, // number of blocks in the experiment
+    trials_per_block = 10, // number of trials per block
+    trial_timeout = 15000, // timeout per trial in milliseconds (15 seconds default)
+    enable_timeout = true, // whether to enable auto timeout
+    // ===== TEXT-TO-SPEECH CONFIGURATION =====
+    enable_tts = true, // Enable text-to-speech functionality
+    tts_method = 'google', // Preferred TTS method: 'google', 'system'
+    tts_rate = 1.0, // Speech rate (0.1 to 10)
+    tts_pitch = 1.0, // Speech pitch (0 to 2)
+    tts_volume = 1.0, // Speech volume (0 to 1)
+    tts_lang = 'en-US', // Language code for TTS
+
+} : {
+    max_pumps?: number,
+    min_pumps?: number,
+    currency_unit_per_pump?: number, // eg 1 cent per pump
+    num_blocks?: number, // number of blocks in the experiment
+    trials_per_block?: number, // number of trials per block
+    trial_timeout?: number, // timeout per trial in milliseconds
+    enable_timeout?: boolean, // whether to enable auto timeout
+    // TTS Configuration
+    enable_tts?: boolean, // Enable text-to-speech functionality
+    tts_method?: 'google' | 'system', // Preferred TTS method (default: 'google')
+    tts_rate?: number, // Speech rate (0.1 to 10, default: 1.0)
+    tts_pitch?: number, // Speech pitch (0 to 2, default: 1.0)
+    tts_volume?: number, // Speech volume (0 to 1, default: 1.0)
+    tts_lang?: string, // Language code for TTS
+} = {})
+{ 
+    //jsPsych = jsPsych;
+
+    // Create TTS options object
+    const ttsOptions = {
+        method: tts_method,
+        rate: tts_rate,
+        pitch: tts_pitch,
+        volume: tts_volume,
+        lang: tts_lang
+    };
+
+    const trial = createTrialTimeline(jsPsych, max_pumps, min_pumps, currency_unit_per_pump, trial_timeout, enable_timeout);
+    
+    // Create block structure
+    const blocks = [];
+    for (let block = 1; block <= num_blocks; block++) {
+        // Add trials for this block
+        const blockTimeline = {
+            timeline: [trial],
+            repetitions: trials_per_block,
+        };
+        blocks.push(blockTimeline);
+        
+        // Add break screen between blocks (but not after the last block)
+        if (block < num_blocks) {
+            const blockBreak = showBlockBreak(jsPsych, block, num_blocks, enable_tts, ttsOptions);
+            blocks.push(blockBreak);
+        }
+    }
+    
+    const bart_timeline = {
+        timeline: blocks
+    }
+     return bart_timeline;;
+
+};
+export const timelineUnits = {
+    createTrialTimeline
+}
+
+export const utils = {
```

Comment:
I think in general the `utils` should be things that don't return trials/timelines. I'd move this stuff to `timelineUnits`